src/JustEat.StatsD/V2/PooledUdpTransportV2.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using JustEat.StatsD.EndpointLookups;
+
+namespace JustEat.StatsD.V2
+{
+    public sealed class PooledUdpTransportV2 : IDisposable, IStatsDTransportV2
```

Comment:
Why not make `PooledUdpTransport` just implement `IStatsDTransport` _and_ `IStatsDTransportV2`? Then we don't have two types with almost the same implementation.